cmd/gapit/replace_resource.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (C) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/gapid/core/app"
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/service"
+)
+
+type replaceResourceVerb struct{ ReplaceResourceFlags }
+
+func init() {
+	verb := &replaceResourceVerb{
+		ReplaceResourceFlags{
+			At:              -1,
+			OutputTraceFile: "newcapture.gfxtrace",
+		},
+	}
+	app.AddVerb(&app.Verb{
+		Name:      "replace_resource",
+		ShortHelp: "Produce a new trace with the given resource replaced at the given command",
+		Action:    verb,
+	})
+}
+
+func (verb *replaceResourceVerb) Run(ctx context.Context, flags flag.FlagSet) error {
+	if flags.NArg() != 3 {
+		app.Usage(ctx, "Exactly 3 arguments: resource handle, resource path, capture, got %d", flags.NArg())
+		return nil
+	}
+	resourceHandle := flags.Arg(0)
```

Comment:
Nit: We typically use flags for non-capture gapit parameters (keeping the non-flag args to 1). Fine to keep as-is, but be aware that we might change this later.

Reply:
That's what I wrote initially, then changed, cause I thought flags are the optional arguments. Changed it back, to follow the convention. The resourcePath and handle are required however, I didn't see a pattern in gapit words for required flags.